governance/program/src/state.rs

```diff
@@ -0,0 +1,281 @@
+//! Program accounts
+
+use solana_program::{epoch_schedule::Slot, instruction::Instruction, pubkey::Pubkey};
+
+/// Max number of instructions allowed for a proposal
+pub const MAX_INSTRUCTIONS: usize = 5;
+
+/// Defines all Governance accounts types
+#[derive(Clone, Debug, PartialEq)]
+pub enum GovernanceAccountType {
+    /// Default uninitialized account state
+    Uninitialized,
+
+    /// Top level aggregation for governances within Governance Token (and optional Council Token).
+    Realm,
+
+    /// Voter record for each voter within a Realm.
+    VoterRecord,
+
+    /// Program Governance account.
+    ProgramGovernance,
+
+    /// Proposal account for Governance account. A single Governance account can have multiple Proposal accounts.
+    Proposal,
+
+    /// Proposal voting state account. Every Proposal account has exactly one ProposalState account.
+    ProposalState,
+
+    /// Vote record account for a given Proposal.  Proposal can have 0..n voting records.
+    ProposalVoteRecord,
+
+    /// Single Signer Instruction account which holds an instruction to execute for Proposal.
+    SingleSignerInstruction,
+}
+
+impl Default for GovernanceAccountType {
+    fn default() -> Self {
+        GovernanceAccountType::Uninitialized
+    }
+}
+
+/// Vote  with number of votes
+#[derive(Clone, Debug, PartialEq)]
+pub enum VoteWeight {
+    /// Yes vote
+    Yes(u64),
+
+    /// No vote
+    No(u64),
+}
+
+/// Yes/No Vote
+#[derive(Clone, Debug, PartialEq)]
+pub enum Vote {
```

Comment:
nit: I think this enum now lives in instruction.rs, it's only used by `GovernanceInstruction::Vote`

Reply:
Moved